core/src/main/kotlin/org/evomaster/core/search/gene/sql/SqlMultidimensionalArrayGene.kt

```diff
@@ -0,0 +1,348 @@
+package org.evomaster.core.search.gene.sql
+
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.search.StructuralElement
+import org.evomaster.core.search.gene.ArrayGene
+import org.evomaster.core.search.gene.Gene
+import org.evomaster.core.search.gene.GeneUtils
+import org.evomaster.core.search.service.Randomness
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * https://www.postgresql.org/docs/14/arrays.html
+ * A multidimensional array representation.
+ * When created, all dimensions have length 0
+ */
+class SqlMultidimensionalArrayGene<T>(
```

Comment:
@ZhangMan1126  can you look at this class?
when we add new genes (which might happen again in the future) we need to make sure that we handle all of their features, including adaptive hypermutation. 
for example, does this implementation handle it? if not, what is missing? likely we would need to write some documentation and/or test for when we add new genes

Reply:
@arcuri82 it does not enable the adaptive hypermutation, we need to add corresponding impacts and override methods of `Gene`. I will write such documentation. where do we put the documentation? part of README (eg for developers) or comments in `Gene`?

Reply:
thx. let's have it in `Gene`. then, @jgaleotti can try to follow such documentation, and see if he can implement this gene correctly

Reply:
the documentation is added in the 32ecdc5 commit. let me know if anything is unclear @jgaleotti

Reply:
ok. I will have a look at it.